tools/build-for-arm.sh

```diff
@@ -0,0 +1,84 @@
+#!/bin/bash
+#
+# Copyright 2021 Istio Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+set -ex
+
+CURR_DIR=$(dirname "${BASH_SOURCE[0]}")
+
+if [ ! -d ${CURR_DIR}/istio ]; then
+    git clone https://github.com/istio/istio.git &&
+    git clone https://github.com/istio/proxy.git &&
+    git clone https://github.com/istio/tools.git
+fi
+
+# Build build-tools and build-tools-proxy
+pushd ${CURR_DIR}/tools/docker/build-tools
+if [ -z "$(docker images |grep "build-tools-proxy" | grep "master-latest")" ]; then
+    DRY_RUN=1 time ./build-and-push.sh
```

Comment:
Seems like we should assert users have HUB set to something? Otherwise it will default to Istio official GCR which will either fail or if some devs have permission unexpectedly push

Reply:
Yes, we just use the default value of "gcr.io/istio-testing" to build an image named as gcr.io/istio-testing/build-tools-proxy:master-latest locally and use it the base building image for other images.

Reply:
Currently I think we can just choose the default one because the following building would rely on this image, or we need to explicitly set the IMG variable for subsequent building. 
I can do it subsequently if it's actual needed. But currently it works fine for a local building on arm.